File: math/src/test/kotlin/dev/peytob/math/matrix/Mat4FTest.java
package dev.peytob.math.matrix;

import org.junit.jupiter.api.Test;

import static dev.peytob.math.matrix.MatricesKt.immutableMat4F;
import static org.junit.jupiter.api.Assertions.*;

class Mat4FTest {

    @Test
    void equalsMethodTest() {
        Mat4 first = immutableMat4F(
            1, 2, 3, 4,
            5, 6, 7, 8,
            9, 0, 1, 2,
            3, 4, 5, 6
        );

        Mat4 second = immutableMat4F(
            1, 2, 3, 4,
            5, 6, 7, 8,
            9, 0, 1, 2,
            3, 4, 5, 6
        );

        assertEquals(first, second);
    }
}

File: math/src/test/kotlin/dev/peytob/math/matrix/Mat4fTest.kt
package dev.peytob.math.matrix

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class Mat4fTest {

    @Test
    fun equalsMethodTest() {
        val first = immutableMat4f(
            1f, 2f, 3f, 4f,
            5f, 6f, 7f, 8f,
            9f, 0f, 1f, 2f,
            3f, 4f, 5f, 6f
        )

        val second = immutableMat4f(
            1f, 2f, 3f, 4f,
            5f, 6f, 7f, 8f,
            9f, 0f, 1f, 2f,
            3f, 4f, 5f, 6f
        )

        assertEquals(first, second)
    }
}